### ISSUE TITLE
SizedOutput::new (` MmapOptions::new().map_mut`) fails for `/dev/null`

### ISSUE THREAD

**Issue by marxin** [davidlattimore/wild#96]
```
❯ echo 'int main() {}' | gcc -B ~/Programming/wild -static -o /dev/null -xc -
Error: Invalid argument (os error 22)
collect2: error: ld returned 1 exit status
```

**Comment by davidlattimore** [comment]
At first I thought perhaps linkers might support output to arbitrary devices, so I tried outputting to `/dev/stdout`, but GNU ld failed saying that it couldn't seek. Do you think we just special-case `/dev/null` and skip writing the output?

**Comment by marxin** [comment]
I would rather allocate a `Vec<u8>` buffer that will be used instead of the mmaped output file. Doing that, the entire linker functionality will be tested and it might help isolate filesystem (disk) speed when it comes to performance measurements.

**Comment by davidlattimore** [comment]
Yeah, you're right, that probably is the best option. Just because writing to `/dev/stdout` doesn't work with GNU ld, doesn't mean we need to go out of our way to make it not work with Wild.

**Comment by davidlattimore** [comment]
I'll implement this now
